csi.proto

```diff
@@ -503,6 +485,147 @@ message ControllerServiceCapability {
     RPC rpc = 1;
   }
 }
+message CreateSnapshotRequest {
+  // The ID of the source volume to be snapshotted.
+  // This field is REQUIRED.
+  string source_volume_id = 1;
+
+  // The suggested name for the snapshot. This field is REQUIRED for
+  // idempotency.
+  string name = 2;
```

Comment:
So here in CreateSnapshotRequest, we are REQUIRED to pass a `name`. and this better be a fresh one or the snapshot does not get created. Right? 

If that's the case, we should be able to query the `name`(s) we used before .... However, `ListSnapshotsResponse` has a list of `Snapshot` object, BUT these objects do not have the `name` that was used when they were created.

So how do we keep track of the names that were already used?

Reply:
@dims The name in CreateSnapshotRequest works the same way as the name in CreateVolumeRequest.  In create volume, name is unique and it will be saved in PV in K8S so we can find a mapping between the volume name and the id generated by the plugin in the PV source.  CreateSnapshot will work the same way.  Snapshot will get an ID after it is created by the plugin and when we call ListSnapshots we pass in the snapshot ID to find it.

Reply:
@xing-yang the code that created the snapshot is the ONLY one who knows which name is mapped to which ID. in create volumes like you mentioned anyone can look in the PV to find the name and ID. that's not the case here

Reply:
@dims For snapshot, we have VolumeSnapshot which is equivalent to PVC and VolumeSnapshotData which is equivalent to PV.  We also have data source for snapshot, similar to the data source for PV.  However those design info will be in the design proposal for K8S, not here:).  This is a CSI spec that involves other CO's in addition to K8S.

Reply:
@dims I heard you.  Can you take a look of the "RPC Interactions" section?  It explains that name may be different from resource id.  Maybe you'll change your mind after reading that:)?

Reply:
From the "RPC Interactions" section, you can see that it was a design decision that the CO-generated volume `name` may be different from the plugin-generated `volume_id`.  This PR for snapshot is consistent with that design.